File: template-project-microservice/template-project-frontend/src/services/product-service.js
/**
 * 商品服务
 */

import productApi from '@api/product.js';
import logger from '@utils/logger.js';

class ProductService {
  /**
   * 获取商品列表
   */
  async getProducts(params = {}) {
    try {
      const response = await productApi.getProducts(params);
      if (response.data && response.data.list) {
        return {
          list: response.data.list,
          total: response.data.total || 0,
          pageNum: response.data.pageNum || 1,
          pageSize: response.data.pageSize || 10,
          pages: response.data.pages || 0
        };
      }
      return {
        list: response.data || [],
        total: response.data?.length || 0,
        pageNum: 1,
        pageSize: 10,
        pages: 1
      };
    } catch (error) {
      logger.error('Failed to get products', error);
      throw error;
    }
  }

  /**
   * 获取商品详情
   */
  async getProductById(id) {
    try {
      const response = await productApi.getProductById(id);
      return response.data || response;
    } catch (error) {
      logger.error('Failed to get product', error);
      throw error;
    }
  }

  /**
   * 创建商品
   */
  async createProduct(data) {
    try {
      const response = await productApi.createProduct(data);
      logger.info('Product created', { id: response.data?.id });
      return response.data || response;
    } catch (error) {
      logger.error('Failed to create product', error);
      throw error;
    }
  }

  /**
   * 更新商品
   */
  async updateProduct(id, data) {
    try {
      const response = await productApi.updateProduct(id, data);
      logger.info('Product updated', { id });
      return response.data || response;
    } catch (error) {
      logger.error('Failed to update product', error);
      throw error;
    }
  }

  /**
   * 删除商品
   */
  async deleteProduct(id) {
    try {
      await productApi.deleteProduct(id);
      logger.info('Product deleted', { id });
    } catch (error) {
      logger.error('Failed to delete product', error);
      throw error;
    }
  }

  /**
   * 获取商品类型列表
   */
  async getProductTypes() {
    try {
      const response = await productApi.getProductTypes();
      return response.data || [];
    } catch (error) {
      logger.error('Failed to get product types', error);
      return [];
    }
  }
}

export default new ProductService();

File: template-project-microservice/template-project-cloud/cloud-modules/product-service/src/main/java/com/example/product/application/service/ProductService.java
package com.example.product.application.service;

import com.example.core.common.exception.BusinessException;
import com.example.core.common.result.ResultCode;
import com.example.product.application.dto.CreateProductCommand;
import com.example.product.application.dto.PageResult;
import com.example.product.application.dto.ProductDTO;
import com.example.product.application.dto.UpdateProductCommand;
import com.example.product.domain.entity.ProductInfo;
import com.example.product.domain.repository.ProductInfoRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.util.StringUtils;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

/**
 * 商品服务
 */
@Service
@RequiredArgsConstructor
public class ProductService {
    private final ProductInfoRepository productInfoRepository;

    public PageResult<ProductDTO> getProductList(int pageNum, int pageSize, String keyword) {
        List<ProductInfo> products = productInfoRepository.findList(pageNum, pageSize, keyword);
        long total = productInfoRepository.count(keyword);

        List<ProductDTO> productDTOs = products.stream()
                .map(this::convertToDTO)
                .collect(Collectors.toList());

        return new PageResult<>(productDTOs, total, pageNum, pageSize);
    }

    public ProductDTO getProductById(String id) {
        ProductInfo product = productInfoRepository.findById(id);
        if (product == null) {
            throw new BusinessException(ResultCode.NOT_FOUND, "商品不存在");
        }
        return convertToDTO(product);
    }

    @Transactional
    public ProductDTO createProduct(CreateProductCommand command) {
        ProductInfo product = new ProductInfo();
        product.setProductName(command.getProductName());
        product.setProductCode(command.getProductCode());
        product.setTypeId(command.getTypeId());
        product.setPrice(command.getPrice());
        product.setStock(command.getStock());
        product.setDescription(command.getDescription());
        product.setEnabled(command.getStatus() == 1);
        product.setCreateDate(LocalDateTime.now());
        product.setCreateUser("system");
        product.setDeleted(0);
        product.setDbVersion(1);

        productInfoRepository.save(product);
        return convertToDTO(product);
    }

    @Transactional
    public ProductDTO updateProduct(String id, UpdateProductCommand command) {
        ProductInfo product = productInfoRepository.findById(id);
        if (product == null) {
            throw new BusinessException(ResultCode.NOT_FOUND, "商品不存在");
        }

        if (StringUtils.hasText(command.getProductName())) {
            product.setProductName(command.getProductName());
        }
        if (StringUtils.hasText(command.getProductCode())) {
            product.setProductCode(command.getProductCode());
        }
        if (StringUtils.hasText(command.getTypeId())) {
            product.setTypeId(command.getTypeId());
        }
        if (command.getPrice() != null) {
            product.setPrice(command.getPrice());
        }
        if (command.getStock() != null) {
            product.setStock(command.getStock());
        }
        if (command.getDescription() != null) {
            product.setDescription(command.getDescription());
        }
        if (command.getStatus() != null) {
            product.setEnabled(command.getStatus() == 1);
        }

        product.setUpdateDate(LocalDateTime.now());
        product.setUpdateUser("system");
        product.setDbVersion(product.getDbVersion() + 1);

        productInfoRepository.update(product);
        return convertToDTO(product);
    }

    @Transactional
    public void deleteProduct(String id) {
        ProductInfo product = productInfoRepository.findById(id);
        if (product == null) {
            throw new BusinessException(ResultCode.NOT_FOUND, "商品不存在");
        }
        productInfoRepository.delete(id);
    }

    private ProductDTO convertToDTO(ProductInfo product) {
        ProductDTO dto = new ProductDTO();
        dto.setId(product.getId());
        dto.setProductName(product.getProductName());
        dto.setProductCode(product.getProductCode());
        dto.setTypeId(product.getTypeId());
        dto.setCategory(product.getTypeId()); // 暂时用typeId作为category
        dto.setPrice(product.getPrice());
        dto.setStock(product.getStock());
        dto.setDescription(product.getDescription());
        dto.setImageUrl(product.getImageUrl());
        dto.setStatus(product.getEnabled() ? 1 : 0);
        dto.setCreateDate(product.getCreateDate());
        dto.setUpdateDate(product.getUpdateDate());
        return dto;
    }
}



